crates/core/component/stake/src/component.rs

```diff
@@ -324,22 +326,61 @@ pub(crate) trait StakingImpl: StateWriteExt {
                     .push(u);
             }
         }
+
         tracing::debug!(
             total_delegations = ?delegations_by_validator.values().map(|v| v.len())
                 .sum::<usize>(),
             total_undelegations = ?undelegations_by_validator.values().map(|v| v.len())
                 .sum::<usize>(),
+                epoch_start_height = epoch_to_end.start_height,
+                epoch_end_height = end_height,
+                "calculated delegation changes for epoch"
         );
 
-        let chain_params = self.get_stake_params().await?;
-
         // We are transitioning to the next epoch, so the "current" base rate in
         // the state is now the previous base rate.
         let prev_base_rate = self.current_base_rate().await?;
-        // TODO: This will change to be based on issuance.
+
+        tracing::debug!(
+            "fetching the issuance budget for this epoch from the distributions component"
+        );
+
+        // Fetch the issuance budget for the epoch we are ending.
+        let issuance_budget_for_epoch = self
+            .get_staking_token_issuance_for_epoch()
+            .expect("issuance budget is always set by the distributions component");
+
+        tracing::debug!(
+            ?issuance_budget_for_epoch,
+            "calculating base rate for next epoch from issuance budget"
+        );
+
+        // Compute the base reward rate for the upcoming epoch based on the total amount
+        // of active stake and the issuance budget given to us by the distribution component.
         tracing::debug!("processing base rate");
-        let next_base_rate = prev_base_rate.next(chain_params.base_reward_rate);
-        tracing::debug!(?prev_base_rate, ?next_base_rate);
+        let bps_squared = Amount::from(1_0000_0000u128);
+        let issuance_budget_for_epoch_bps = issuance_budget_for_epoch * bps_squared;
+
+        let total_active_stake_previous_epoch = self.total_active_stake().await?;
+        let base_reward_rate: Amount = U128x128::ratio(
+            issuance_budget_for_epoch_bps,
+            total_active_stake_previous_epoch,
+        )
+        .expect("total active stake is nonzero")
+        .round_down()
+        .try_into()
+        .expect("rounded to an integral value");
+
+        // MERGEBLOCK(erwan): unsafe casting to `u64`. Should we use `Amount`s?
+        let next_base_rate = prev_base_rate.next(base_reward_rate.value() as u64);
+        tracing::debug!(
+            ?prev_base_rate,
+            ?next_base_rate,
+            ?base_reward_rate,
+            ?total_active_stake_previous_epoch,
+            ?issuance_budget_for_epoch,
+            "calculated base rate for the upcoming epoch"
+        );
```

Comment:
I think this shouldn't be merge-blocking, but we should instead fold it into an issue about reworking the staking arithmetic (in general) to use the same `U128x128` codepaths as other arithmetic.